File: CSBMFramework/csbm/src/main/java/com/csbm/BERESTConfigCommand.java
package com.csbm;


import com.csbm.http.BEHttpRequest;

import java.util.HashMap;
import java.util.Map;

/** package */ class BERESTConfigCommand extends BERESTCommand {

  public BERESTConfigCommand(
      String httpPath,
      BEHttpRequest.Method httpMethod,
      Map<String, ?> parameters,
      String sessionToken) {
    super(httpPath, httpMethod, parameters, sessionToken);
  }

  public static BERESTConfigCommand fetchConfigCommand(String sessionToken) {
    return new BERESTConfigCommand("config", BEHttpRequest.Method.GET, null, sessionToken);
  }

  public static BERESTConfigCommand updateConfigCommand(
          final Map<String, ?> configParameters, String sessionToken) {
    Map<String, Map<String, ?>> commandParameters = null;
    if (configParameters != null) {
      commandParameters = new HashMap<>();
      commandParameters.put("params", configParameters);
    }
    return new BERESTConfigCommand(
        "config", BEHttpRequest.Method.PUT, commandParameters, sessionToken);
  }
}

File: BEFramework/BEFramework/Internal/Commands/REST/Command/BERESTConfigCommand.m
//
//  BERESTConfigCommand.m
//  BEFramework
//
//  Created by Loc Nguyen on 5/26/16.
//  Copyright © 2016 Loc Nguyen. All rights reserved.
//

#import "BERESTConfigCommand.h"
#import "BEAssert.h"
#import "BEHTTPRequest.h"

@implementation BERESTConfigCommand

+ (instancetype)configFetchCommandWithSessionToken:(NSString *)sessionToken {
  return [self commandWithHTTPPath:@"config"
                        httpMethod:BEHTTPRequestMethodGET
                        parameters:nil
                      sessionToken:sessionToken];
}

+ (instancetype)configUpdateCommandWithConfigParameters:(NSDictionary *)parameters
                                           sessionToken:(NSString *)sessionToken {
  NSDictionary *commandParameters = @{ @"params" : parameters };
  return [self commandWithHTTPPath:@"config"
                        httpMethod:BEHTTPRequestMethodPUT
                        parameters:commandParameters
                      sessionToken:sessionToken];
}

@end
